packages/flutter/lib/src/widgets/scroll_absolute.dart

```diff
@@ -78,39 +78,69 @@ class ViewportScrollBehavior extends ScrollBehavior2 {
     return null;
   }
 
-  @override
-  ScrollPosition createScrollPosition(BuildContext context, Scrollable2State state, ScrollPosition oldPosition) {
-    switch (getPlatform(context)) {
+  ScrollPhysics _getScrollPhysics(TargetPlatform platform) {
+    assert(platform != null);
+    switch (platform) {
       case TargetPlatform.iOS:
-        return new _CupertinoViewportScrollPosition(
-          state,
-          scrollTolerances,
-          oldPosition,
-        );
+        return const BouncingScrollPhysics();
       case TargetPlatform.android:
       case TargetPlatform.fuchsia:
-        return new _MountainViewViewportScrollPosition(
-          state,
-          scrollTolerances,
-          oldPosition,
-        );
+        return const ClampingScrollPhysics();
     }
     return null;
   }
 
   @override
+  ScrollPosition createScrollPosition(BuildContext context, Scrollable2State state, ScrollPosition oldPosition) {
+    return new AbsoluteScrollPosition(state, scrollTolerances, oldPosition, _getScrollPhysics(getPlatform(context)));
+  }
+
+  @override
   bool shouldNotify(ViewportScrollBehavior oldDelegate) {
     return scrollTolerances != oldDelegate.scrollTolerances;
   }
 }
 
-abstract class AbsoluteScrollPosition extends ScrollPosition {
+abstract class ScrollPhysics {
+  const ScrollPhysics();
+
+  /// Used by [AbsoluteDragScrollActivity] and other user-driven activities to
+  /// convert an offset in logical pixels as provided by the [DragUpdateDetails]
+  /// into a delta to apply using [setPixels].
+  ///
+  /// This is used by some [ScrollPosition] subclasses to apply friction during
+  /// overscroll situations.
+  double applyPhysicsToUserOffset(AbsoluteScrollPosition position, double offset) => offset;
+
+  /// Determines the overscroll by applying the boundary conditions.
+  ///
+  /// Called by [AbsoluteScrollPosition.setPixels] just before the [pixels] value is updated, to
+  /// determine how much of the offset is to be clamped off and sent to
+  /// [AbsoluteScrollPosition.reportOverscroll].
+  ///
+  /// The `value` argument is guaranteed to not equal [pixels] when this is
+  /// called.
+  double applyBoundaryConditions(AbsoluteScrollPosition position, double value) => 0.0;
```

Comment:
It seems incestuous to have this function take a ScrollPosition.

It seems just wrong for it to take an AbsoluteScrollPosition.